### PR TITLE
{math}[foss/2017b] HIPS v1.2b-rc5

### PR DESCRIPTION
**Pull request by smoors**

### CODE REVIEW

**Comment by verdurin** [comment]
@smoors please remove the `SCOTCH` `easyconfig` which has already been merged in https://github.com/easybuilders/easybuild-easyconfigs/pull/5735

**Comment by smoors** [comment]
@wpoely86 review?

File: easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+prebuildopts += 'echo "PARTITIONER = -DSCOTCH_PART" >>makefile.inc && '
+prebuildopts += 'echo "ARCH = -DLINUX" >>makefile.inc && '
+prebuildopts += 'echo "LD = \$(FC)" >>makefile.inc && '
+prebuildopts += 'echo "MPILD = \$(MPIFC)" >>makefile.inc && '
+prebuildopts += 'echo "LBLAS = $LIBBLAS" >>makefile.inc && '
+prebuildopts += 'echo "MAKE = make" >>makefile.inc && '
+prebuildopts += 'echo "AR = ar" >>makefile.inc && '
+prebuildopts += 'echo "ARFLAGS = -crs" >>makefile.inc && '
+prebuildopts += 'echo "LN = ln" >>makefile.inc && '
+prebuildopts += 'echo "CP = cp" >>makefile.inc && '
```
Comment (boegel) [inline easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb]: Hmm, isn't it easier to just copy `makefile.inc.gnu` to `makefile.inc` via `prebuildopts`, and then override values via `buildopts` like this:

```python
buildopts = 'ISCOTCH="-I$EBROOTSCOTCH/include" LSCOTCH="-L$EBROOTSCOTCH/lib -lscotch -lscotcherr" '

Comment (smoors) [reply easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb]: cool, I did not know you could do that

File: easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+prebuildopts = 'cp Makefile_Inc_Examples/makefile.inc.gnu makefile.inc && '
+
+# make all compiles both the library and the test programs
+buildopts = (
+    'LBLAS="$LIBBLAS" '
+    'COEFTYPE=-DTYPE_REAL '
+    'PARTITIONER=-DSCOTCH_PART '
+    'ISCOTCH="-I$EBROOTSCOTCH/include" '
+    'LSCOTCH="-L$EBROOTSCOTCH/lib -lscotch -lscotcherr" '
+    'all '
```
Comment (boegel) [inline easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb]: @smoors We usually format multi-line `buildopts` like this (keeping related options together, use double quotes for all values for consistency):

```python
buildopts = 'all LBLAS="$LIBBLAS" COEFTYPE="-DTYPE_REAL" PARTITIONER="-DSCOTCH_PART" '
buildopts += 'ISCOTCH="-I$EBROOTSCOTCH/include" LSCOTCH="-L$EBROOTSCOTCH/lib -lscotch -lscotcherr"'

File: easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+files_to_copy = [(['LIB/*'], 'lib')]
```
Comment (boegel) [inline easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb]: I also see `*.h` files being copied into `lib`, shouldn't we copy those to `include` instead (and adjust `sanity_check_paths` accordingly)?

```python
files_to_copy = [
    (['LIB/*.h', 'LIB/*.inc'], 'include'),
    (['LIB/*.a'], 'lib'),
]

sanity_check_paths = {
    'files': ['include/hips_fortran.h', 'include/hips.h', 'include/hips.inc', 'include/io.h', 'include/type.h',
              'lib/libhips.a', 'lib/libhipssequential.a', 'lib/libio.a', 'lib/libspkit.a'],
    'dirs': [],
}

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2316.phanpy.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/75d889b8650a8fcdb47caceb1ba848cd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2111.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/92964aa86032c61350a6a875388cc288 for a full test report.

File: easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/h/HIPS/HIPS-1.2b-rc5-foss-2017b.eb]: @smoors What about this?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gligar01.gligar.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4d535ad3efa3be932e714bb55b56fc6d for a full test report.

**Comment by boegel** [comment]
Going in, thanks @smoors!

**Pull request merged**